core/cloudflow-operator/src/test/scala/cloudflow/operator/runner/AkkaRunnerSpec.scala

```diff
@@ -0,0 +1,125 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.operator.runner
+
+import cloudflow.blueprint._
+import cloudflow.blueprint.deployment.{ PrometheusConfig, StreamletDeployment }
+import cloudflow.operator.{ CloudflowApplication, CloudflowApplicationSpecBuilder, TestDeploymentContext }
+import cloudflow.operator.runner.SparkResource.{ AlwaysRestartPolicy, CR }
+import com.typesafe.config.ConfigFactory
+import org.scalatest._
+import play.api.libs.json._
+import skuber._
+
+class AkkaRunnerSpec extends WordSpecLike with OptionValues with MustMatchers with GivenWhenThen with TestDeploymentContext {
+
+  case class Foo(name: String)
+  case class Bar(name: String)
+
+  import BlueprintBuilder._
+  val appId             = "some-app-id"
+  val image             = "docker-registry.foo.com/lightbend/call-record-pipeline:277-ceb9629"
+  val clusterName       = "cloudflow-strimzi"
+  val pvcName           = "my-pvc"
+  val namespace         = "test-ns"
+  val prometheusJarPath = "/app/prometheus/prometheus.jar"
+  val prometheusConfig  = PrometheusConfig("(prometheus rules)")
+
+  "SparkRunner" should {
+
+    val appId      = "some-app-id"
+    val appVersion = "42-abcdef0"
+    val agentPaths = Map(CloudflowApplication.PrometheusAgentKey -> "/app/prometheus/prometheus.jar")
+    val image      = "docker-registry.foo.com/lightbend/call-record-pipeline:277-ceb9629"
+    val namespace  = "test-ns"
+
+    val ingress = randomStreamlet().asIngress[Foo].withServerAttribute
+    val egress  = randomStreamlet().asEgress[Foo].withServerAttribute
+
+    val ingressRef = ingress.ref("ingress")
+    val egressRef  = egress.ref("egress")
+
+    val verifiedBlueprint = Blueprint()
+      .define(Vector(ingress, egress))
+      .use(ingressRef)
+      .use(egressRef)
+      .connect(Topic("foos"), ingressRef.out, egressRef.in)
+      .verified
+      .right
+      .value
+
+    val app = CloudflowApplication(CloudflowApplicationSpecBuilder.create(appId, appVersion, image, verifiedBlueprint, agentPaths))
+
+    val deployment = StreamletDeployment(
+      name = appId,
+      runtime = "spark",
+      image = image,
+      streamletName = "spark-streamlet",
+      className = "cloudflow.operator.runner.AkkaRunner",
+      endpoint = None,
+      secretName = "spark-streamlet",
+      config = ConfigFactory.empty(),
+      portMappings = Map.empty,
+      volumeMounts = None,
+      replicas = None
+    )
+
+    val volumes = List(
```

Comment:
The volumes and volumeMounts are not used in the test I think?

Reply:
very right